### PR TITLE
Adding hotstarting, gradient-only condensing and primal-only expansion for qpoases - bumping hpipm

### PR DESCRIPTION
**Pull request by zanellia**
Adding hotstarting, gradient-only condensing and primal-only expansion for qpoases. This can be useful for linear MPC and inexact schemes with frozen sensitivities. The pull request also bumps hpipm since primal-only expansion has been added.

### CODE REVIEW

File: acados/dense_qp/dense_qp_qpoases.c [hunk 1]
```diff
@@ -81,6 +82,7 @@ int dense_qp_qpoases_calculate_memory_size(dense_qp_dims *dims, void *args_)
     int size = sizeof(dense_qp_qpoases_memory);
 
     size += 1 * nvd * nvd * sizeof(double);        // H
+	size += 1 * nvd * nvd * sizeof(double);        // R
```
Comment (dkouzoup) [inline acados/dense_qp/dense_qp_qpoases.c]: Stupid question: Do you have 8 spaces for tab or is it shown wrong throughout the PR for a reason?

File: acados/dense_qp/dense_qp_qpoases.c [hunk 2]
```diff
@@ -65,6 +65,7 @@ void dense_qp_qpoases_initialize_default_args(void *args_)
     args->max_cputime = 1000.0;
     args->warm_start = 0;
     args->max_nwsr = 1000;
+	/* args->use_precomputed_choleski = 0; */
```
Comment (dkouzoup) [inline acados/dense_qp/dense_qp_qpoases.c]: 1. use_precomputed_cholesk**y** (all occurrences)
2. Should you not initialize both hotstart and use_precomputed_cholesky to zero here?

File: acados/ocp_qp/ocp_qp_hpmpc.c [hunk 3]
```diff
@@ -268,18 +268,12 @@ int ocp_qp_hpmpc(ocp_qp_in *qp_in, ocp_qp_out *qp_out, void *args_, void *mem_,
 	blasfeo_dvecsc(ng[ii], -1.0, &hsd[ii], 2*nb[ii] + ng[ii]);
    	
 	// dmat loop	
-	for ( ii = 0; ii < N; ii++ ) {
-
-        /* blasfeo_create_dmat(nu[ii]+nx[ii]+1, nu[ii]+nx[ii], &hsL[ii], ptr_memory); */
```
Comment (dkouzoup) [inline acados/ocp_qp/ocp_qp_hpmpc.c]: why ansi-c comments? :)

File: examples/c/mass_spring_offline_fcond_qpoases_split.c [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    ocp_qp_out *qp_out = create_ocp_qp_out(qp_in->dim);
+
+    /************************************************
+    * dense sol
+    ************************************************/
+
+    dense_qp_out *qpd_out = create_dense_qp_out(&ddims);
+
+    /************************************************
+    * dense ipm
```
Comment (dkouzoup) [inline examples/c/mass_spring_offline_fcond_qpoases_split.c]: not ipm

File: acados/dense_qp/dense_qp_qpoases.c (hunk 2 above)
Comment (zanellia) [reply acados/dense_qp/dense_qp_qpoases.c]: Totally. fixed.

File: acados/dense_qp/dense_qp_qpoases.c (hunk 1 above)
Comment (zanellia) [reply acados/dense_qp/dense_qp_qpoases.c]: It's the PR, I believe.

File: acados/ocp_qp/ocp_qp_hpmpc.c (hunk 3 above)
Comment (zanellia) [reply acados/ocp_qp/ocp_qp_hpmpc.c]: Whoops, that's my vim plugin.

File: examples/c/mass_spring_offline_fcond_qpoases_split.c (hunk 4 above)
Comment (zanellia) [reply examples/c/mass_spring_offline_fcond_qpoases_split.c]: fixed too.

**Comment by zanellia** [comment]
By the way, let's wait to merge this. We found a bug in the gradient-only condensing routines with @giaf.

**Comment by zanellia** [comment]
@giaf fixed a bug in  the symv_l BLASFEO that affected the gradient-only condensing and expansion routines (https://github.com/giaf/blasfeo/commit/c5d3351b59895d765c8a4fab19b6155727a03c7d and https://github.com/giaf/blasfeo/commit/b6dc03f79ff9512cf7e6fdbb2c66f9b298b12f86). I guess we can merge this now?

**Pull request merged**